xknx/cemi/cemi_frame.py

```diff
@@ -195,58 +220,459 @@ def from_knx_data_link_layer(cemi: bytes, code: CEMIMessageCode) -> CEMIFrame:
         )
 
+    def __repr__(self) -> str:
+        """Return object as readable string."""
+        return (
+            "CEMILData("
+            f'src_addr="{self.src_addr.__repr__()}" '
+            f'dst_addr="{self.dst_addr.__repr__()}" '
+            f'flags="{self.flags:16b}" '
+            f'tpci="{self.tpci}" '
+            f'payload="{self.payload}")'
+        )
+
+
+class CEMIMPropInfo:
+    """Representation of CEMI Device Management Property."""
+
+    LENGTH = 6
+
+    def __init__(
+        self,
+        *,
+        object_type: ResourceObjectType,
+        object_instance: int = 1,
+        property_id: ResourcePropertyId | int,
+        number_of_elements: int = 1,
+        start_index: int = 1,
+    ):
+        """Initialize CEMIMProp object."""
+        self.object_type = object_type
+        self.object_instance = object_instance
+        self.property_id = (
+            property_id.value
+            if isinstance(property_id, ResourcePropertyId)
+            else property_id
+        )
+        self.number_of_elements = number_of_elements
+        self.start_index = start_index
+
+    def calculated_length(self) -> int:
+        """Get length of CEMI data."""
+        return CEMIMPropInfo.LENGTH
+
+    @staticmethod
+    def from_knx(raw: bytes) -> CEMIMPropInfo:
+        """Parse/deserialize from KNX/IP raw data."""
+        if len(raw) != CEMIMPropInfo.LENGTH:
+            raise CouldNotParseCEMI(
+                f"Invalid CEMI length: {len(raw)}; CEMI: {raw.hex()}"
+            )
+
+        try:
+            object_type = ResourceObjectType(int.from_bytes(raw[0:2], "big"))
+        except ValueError:
+            raise UnsupportedCEMIMessage(
+                f"CEMIMProp Object Type not supported: {raw[0:2].hex()} in CEMI: {raw.hex()}"
+            )
+
+        return CEMIMPropInfo(
+            object_type=object_type,
+            object_instance=raw[2],
+            property_id=raw[3],
+            number_of_elements=(raw[4] >> 4),
+            start_index=(int.from_bytes(raw[4:6], "big") % 0x1000),
+        )
+
     def to_knx(self) -> bytes:
-        """Serialize to KNX/IP raw data."""
-        if self.tpci.control:
-            tpdu = self.tpci.to_knx().to_bytes(1, "big")
-            npdu_len = 0
-        else:
-            if not isinstance(self.payload, APCI):
-                raise ConversionError(
-                    f"Invalid payload set for data TPDU: {type(self.payload)}"
-                )
-            tpdu = self.payload.to_knx()
-            tpdu[0] |= self.tpci.to_knx()
-            npdu_len = self.payload.calculated_length()
+        """Serialize to CEMI raw data."""
+        return (
+            self.object_type.value.to_bytes(2, "big")
+            + self.object_instance.to_bytes(1, "big")
+            + self.property_id.to_bytes(1, "big")
+            + ((self.number_of_elements << 12) + self.start_index).to_bytes(2, "big")
+        )
 
+    def _property_repr(self) -> str:
```

Comment:
why not use `__str__` or `__repr__`?
We could use f"{self.property_info}" instead of `f"{self.property_info._property_repr()}"` below then.